app/controllers/api_controller/authentication.rb

```diff
@@ -52,6 +54,15 @@ def userid_to_userobj(userid)
       User.find_by_userid(userid)
     end
 
+    def authorize_user_group(user_obj)
+      group_name = request.env['HTTP_X_MIQ_GROUP']
+      if group_name.present?
+        group_obj = user_obj.miq_groups.find_by_description(group_name)
+        raise AuthenticationError, "Invalid Authorization Group #{group_name} specified" if group_obj.nil?
+        user_obj.current_group = group_obj
```

Comment:
Think you can just do

```ruby
user_obj.miq_group_description = request.env['HTTP_X_MIQ_GROUP']
```

A little later, I'm going to add the ability for super admin to set a group that they technically aren't in.